testsuites/syncgateway/functional/tests/test_log_redaction.py

```diff
@@ -0,0 +1,454 @@
+import os
+import shutil
+import time
+import subprocess
+import re
+import pytest
+
+from subprocess import Popen, PIPE
+from keywords.ClusterKeywords import ClusterKeywords
+from keywords.exceptions import LogScanningError, CollectionError
+from keywords.SyncGateway import sync_gateway_config_path_for_mode, get_sync_gateway_version
+from keywords.utils import log_info, host_for_url
+from libraries.testkit.cluster import Cluster
+from utilities.cluster_config_utils import load_cluster_config_json
+from utilities.cluster_config_utils import persist_cluster_config_environment_prop, copy_to_temp_conf
+from utilities.scan_logs import scan_for_pattern
+from keywords.MobileRestClient import MobileRestClient
+from keywords import document, attachment
+from libraries.provision.ansible_runner import AnsibleRunner
+
+
+@pytest.mark.sanity
+@pytest.mark.syncgateway
+@pytest.mark.logredaction
+@pytest.mark.parametrize("sg_conf_name, redaction_level", [
+    ("log_redaction", "partial"),
+    ("log_redaction", "none")
+])
+def test_log_redaction_config(params_from_base_test_setup, remove_tmp_sg_redaction_logs, sg_conf_name, redaction_level):
+    """
+    @summary
+    1. Have sync_gateway config file with logging level as partial/none
+    2. Restart sync_gateway
+    3. Create user in sync_gateway
+    4. Create docs with xattrs
+    5. Verify data which is partial sensitive is tagged with 'ud' tags for partial
+    and no ud tags if it is configured with None
+    """
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    cluster_helper = ClusterKeywords(cluster_config)
+    cluster_hosts = cluster_helper.get_cluster_topology(cluster_config)
+    sg_admin_url = cluster_hosts["sync_gateways"][0]["admin"]
+    sg_url = cluster_hosts["sync_gateways"][0]["public"]
+    sg_ip = host_for_url(sg_admin_url)
+    sg_db = "db"
+    num_of_docs = 10
+
+    if get_sync_gateway_version(sg_ip)[0] < "2.1":
+        pytest.skip("log redaction feature not available for version < 2.1 ")
+
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    log_info("Using cluster_conf: {}".format(cluster_config))
+    log_info("Using sg_conf: {}".format(sg_conf))
+
+    # Modifying log redaction level to partial
+    temp_cluster_config = copy_to_temp_conf(cluster_config, mode)
+    print "added redact level in config .... "
+    persist_cluster_config_environment_prop(temp_cluster_config, 'redactlevel', redaction_level, property_name_check=False)
+    cluster = Cluster(config=temp_cluster_config)
+    cluster.reset(sg_config_path=sg_conf)
+
+    # 3. Create user in sync_gateway
```

Comment:
may be change the no.s or remove it. It confuses reader as 1 and 2 are not available in comments. Same for below comments

Reply:
done